---
layout: archive
title: "Publications"
permalink: /publications/
author_profile: true
---

You can find a full list of my publications on <a href="https://scholar.google.com/citations?hl=en&user=pO3M3xYAAAAJ">my Google Scholar profile</a>.

<small>\* = equal contribution.</small>

{% include base_path %}

[**BEND: Benchmarking DNA Language Models on biologically meaningful tasks**](https://arxiv.org/abs/2311.12570)  
<small><b>ICLR, 2024</b></small>  
<small>Frederikke Isa Marin*, <u>Felix Teufel*</u>, Marc Horlacher, Dennis Madsen, Dennis Pultz, Ole Winther, Wouter Boomsma</small>  
We introduce BEND, a collection of prediction tasks for evaluating the performance of DNA LM representations.  
[Code on Github](https://github.com/frederikkemarin/BEND)  
[Data](https://sid.erda.dk/cgi-sid/ls.py?share_id=aNQa0Oz2lY)  

[**SecretoGen: towards prediction of signal peptides for efficient protein secretion**](https://openreview.net/forum?id=vXXEfmYsvS)  
<small><b>GenBio Workshop at NeurIPS, 2023</b></small>  
<small> <u>Felix Teufel</u>, Carsten Stahlhut, Jan Refsgaard, Henrik Nielsen, Ole Winther, Dennis Madsen</small>  
SecretoGen is a generative transformer that designs host- and protein-optimized signal peptides. We show that it can be used for ranking signal peptides sequences that have good secretion performance.  
[Code on Github](https://github.com/fteufel/SecretoGen)

[**GraphPart: homology partitioning for biological sequence analysis**](https://academic.oup.com/nargab/article/5/4/lqad088/7318077)   
<small><b>NAR Genomics and Bioinformatics, 2023</b></small>    
<small> <u>Felix Teufel</u>, Magnús Halldór Gíslason, José Juan Almagro Armenteros, Alexander Rosenberg Johansen, Ole Winther, Henrik Nielsen</small>  
We introduce GraphPart, an algorithm for homology partitioning of biological sequence datasets for machine learning.  
[Code on Github](https://github.com/graph-part/graph-part)  
[Python package](https://pypi.org/project/graph-part/)

[**DeepPeptide predicts cleaved peptides in proteins using conditional random fields**](https://doi.org/10.1093/bioinformatics/btad616)  
<small><b>Bioinformatics, 2023</b></small>  
<small><u>Felix Teufel</u>, Jan C. Refsgaard, Christian T. Madsen, Carsten Stahlhut, Mads Grønborg, Ole Winther, Dennis Madsen</small>  
Predicting native cleaved peptide products directly from precursor protein sequences, without requiring experimental peptide data.  
[Available online at University of Copenhagen Biolib](https://biolib.com/KU/DeepPeptide/)  
[Code on Github](https://github.com/fteufel/deeppeptide/)


[**Deorphanizing Peptides using Structure Prediction**](https://pubs.acs.org/doi/full/10.1021/acs.jcim.3c00378)  
<small><b>Journal of Chemical Information and Modeling, 2023</b> and <b>MLSB Workshop at NeurIPS, 2022</b></small>  
<small><u>Felix Teufel</u>, Jan C. Refsgaard, Marina A. Kasimova, Kristine Deibler, Christian T. Madsen, Carsten Stahlhut, Mads Grønborg, Ole Winther, Dennis Madsen</small>     
We combined AlphaFold-Multimer with DeepTMHMM as a peptide receptor deorphanization method.  
[Code on Github](https://github.com/fteufel/alphafold-peptide-receptors)

[**Combining mass spectrometry and machine learning to discover bioactive peptides**](https://www.nature.com/articles/s41467-022-34031-z)  
<small><b>Nature Communications, 2022</b></small>  
<small>Christian T. Madsen, Jan C. Refsgaard, <u>Felix Teufel</u>, Sonny K. Kjærulff, Zhe Wang, Guangjun Meng, Carsten Jessen, Petteri Heljo, Qunfeng Jiang, Xin Zhao, Bo Wu, Xueping Zhou, Yang Tang, Jacob F. Jeppesen, Christian D. Kelstrup, Stephen T. Buckley, Søren Tullin, Jan Nygaard-Jensen, Xiaoli Chen, Fang Zhang, Jesper V. Olsen, Dan Han, Mads Grønborg & Ulrik de Lichtenberg</small>  
Disovering bioactive peptides in large-scale peptidomics data from their mass spectrometry signal.  
[Code on Github](https://github.com/jancr/ppv) 


**[SignalP 6.0 predicts all five types of signal peptides using protein language models](https://www.nature.com/articles/s41587-021-01156-3)**  
<small><b>Nature Biotechnology, 2022</b></small>  
<small><u>Felix Teufel</u>, Jose Juan Almagro Armenteros, Alexander Rosenberg Johansen,  Magnús Halldór Gíslason, Silas Irby Pihl, Konstantinos D. Tsirigos, Ole Winther, Søren Brunak, Gunnar von Heijne and Henrik Nielsen</small>  
We used protein language models to build SignalP 6.0, the first signal peptide predictor capable of predicting all known types of signal peptides in protein sequences.  
[Available online at DTU Health Tech](https://services.healthtech.dtu.dk/service.php?SignalP-6.0)  
[Code on Github](https://github.com/fteufel/signalp-6.0) 

